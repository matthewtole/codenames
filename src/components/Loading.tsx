import * as React from 'react';
import { Link } from 'react-router-dom';

export const Loading = ({ error }: { error?: string }) => (
  <div className="modal is-active">
    <div className="modal-background" />
    <div className="modal-content has-text-centered has-text-white">
      {error ? (
        <div>
          <i className="fa fa-4x fa-exclamation-triangle" />
          <h2 className="title is-size-3 has-text-white">{error}</h2>
          <h3 className="subtitle is-size-5">
            <Link to="/" className="has-text-light">
              &larr;Back to home
            </Link>
          </h3>
        </div>
      ) : (
        <i className="fa fa-4x fa-spinner fa-pulse" />
      )}
    </div>
  </div>
);
