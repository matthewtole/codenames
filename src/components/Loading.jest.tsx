import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Loading } from './Loading';

describe('<Join />', () => {
  const defaultProps = {};

  it('renders correctly', () => {
    const tree = renderer.create(<Loading {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
