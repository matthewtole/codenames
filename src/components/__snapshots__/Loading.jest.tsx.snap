// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Join /> renders correctly 1`] = `
<div
  className="modal is-active"
>
  <div
    className="modal-background"
  />
  <div
    className="modal-content has-text-centered has-text-white"
  >
    <i
      className="fa fa-4x fa-spinner fa-pulse"
    />
  </div>
</div>
`;
